week1/homework/src/server.js

```diff
@@ -9,7 +9,30 @@ function createServer(port) {
   let state = 10;
 
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    try {
+      switch (request.url) {
+        case '/add':
+          state++;
+          break;
+        case '/subtract':
+          state--;
+          break;
+        case '/reset':
+          state = 10;
+          break;
+      }
+      const stateJSON = { 'state': state };
+      response.setHeader('Content-Type', 'application/json');
+      response.write(JSON.stringify(stateJSON, null, 2));
+    }
+    catch (err) {
+      response.statusCode = 404;
```

Comment:
The code looks good! There is one mistake that's preventing the test to pass. You have a `try/catch` where you are handling errors. In case there is an error you are then returning a `404`. The problem is that the `404` needs to be returned, not when there is an error, but when there are no matches on the URL. 

What's happening now if you go to `http://localhost:3000/blahblah` or any URL that doesn't match is that you are getting the state. Because there are no matches in the `switch` but no errors, so it goes to sending the `state` back. 

I like that you are trying to catch errors. What you need now is a `default` in your `switch` that sends the `404` and then change what you send in the catch. You can send a `500` saying there was an unexpected error. That's not part of the homework though, but since you had the code there already I think is .a nice addition.